codegen/src/pushable.rs

```diff
@@ -89,11 +89,23 @@ fn derive_enum(
             Fields::Unit => quote! { #ident::#variant_ident },
         };
 
-        let push_impl = gen_push_impl(Some(tag), &field_idents, &field_types);
-
-        quote! {
-            #pattern => {
-                #push_impl
+        match &variant.fields {
+            Fields::Named(_) => {
+                let push_impl = gen_push_impl(None, &field_idents, &field_types);
+                quote! {
+                    #pattern => {
+                        #push_impl
+                        ctx.context().push_new_data(vm, #tag as _gluon_types::VmTag, 1)?;
```

Comment:
This should be `push_new_record` instead of a plain data. You will need to create `InternedStr` via `intern` for each field. You can test that it works by passing the record to a row polymorphic function which extracts one of the fields.

```f#
let extract r : { key: String | r } -> String = r.key
```

Reply:
This was correct as it was, my bad. Tested that it worked and fixed the other nits in #773 . Thanks!